### PR TITLE
Utility for computing normal and tangential forces at given plane

### PR DESCRIPTION
**Pull request by lorenzogracia**
New utility for computing normal and tangential forces in the joint element at a given plane.  Also the headers have been updated.

### CODE REVIEW

**Review by ipouplana — CHANGES_REQUESTED** [review]
Hi @lorenzogracia this is a useful utility, but I think it would be better if we could pass the necessary parameters from Python without them being 'hardcoded'...

File: applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        std::cout<< " Normal Force (N) "<<fabs(VectorForceinPlane[2])<<std::endl;
+    }
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+protected:
+
+        /// Member Variables
+        ModelPart& mr_model_part;
+        double mpmid0_0;
```
Comment (rubenzorrilla) [inline applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp]: "mpmid0_0" stands for "member pointer mid0_0" but it is a double (in any case, mid0_0 is not very descriptive...). Besides, might it be possible to store it as a const array type or at least as an const double?.

File: applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        pmid2[2]= mpmid2_2;
+
+        this->CalculateRotationMatrix(RotationPlane,pmid0,pmid1,pmid2);
+        array_1d<double,3> VectorForceinPlane;
+        array_1d<double,3> GlobalElementVectorForce = ZeroVector(3);
+
+        for(int k = 0; k<nelements; k++)
+        {
+            ModelPart::ElementsContainerType::iterator it = el_begin + k;
+            Element::GeometryType& Geom = it->GetGeometry();
```
Comment (rubenzorrilla) [inline applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp]: This is a reference. Call it "rGeom"

File: applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            LocalElementStress[1] *= InvNumGP;
+            LocalElementStress[2] *= InvNumGP;
+            double Area;
+            this->AreaMidPlane(Area,pmid0,pmid1,pmid2);
+            noalias(LocalElementVectorForce) = LocalElementStress*Area;
+            noalias(GlobalElementVectorForce) += prod(trans(RotationMatrix),LocalElementVectorForce);
+        }
+
+        noalias(VectorForceinPlane) = prod(RotationPlane,GlobalElementVectorForce);
+        double TangentialForce = sqrt(VectorForceinPlane[0]*VectorForceinPlane[0] + VectorForceinPlane[1]*VectorForceinPlane[1]);
```
Comment (rubenzorrilla) [inline applications/DamApplication/custom_utilities/global_joint_stress_utility.hpp]: This cant be "const"

**Review by ipouplana — APPROVED** [review]

**Comment by lorenzogracia** [comment]
@rubenzorrilla  thanks for your comments.

**Pull request merged**